Extensions/xkit_main.js

```diff
@@ -12,7 +12,10 @@
 
 		run: function() {
 
-			if (location.href.includes("://www.tumblr.com/login") || location.href.includes("://www.tumblr.com/settings")) {
+			const react = Boolean($("link[href*='/pop/']").length);
```

Comment:
feels like we could just set XKit.page.react up here instead of keeping it as an intermediate variable, but maybe that's too weird. I'm not sure under what circumstances the `if (XKit.page.react === undefined)` check below would ever come into play more than once but maybe git history could enlighten us.....

Reply:
I believe it's the backup "only run this once" condition for the, not-recursive-but-calls-itself "wait for React to load" snippet. No idea who wrote it; I don't recognize the name in the git blame.

```js
if (XKit.page.react) {
  $("body").addClass('xkit--react');
  const waitUntilReactLoaded = setInterval(() => {
    if ($('[data-rh]').length === 0) {
      clearInterval(waitUntilReactLoaded);
      this.run();
    }
  }, 100);
  return;
}
```

On the initial run, if it's a React page but hydration has not yet occurred, this seems to fake an await statement by returning out of the current function invocation and setting an interval timer that calls the whole function from the start again, and setting `XKit.page.react` doubles as a first-time-only check.

Not sure offhand if `run` can just be an async function (probably?); we'd have to write a proper loop or recursive function to clean it up, I guess. XKit Rewritten's version is recursive; I tend to like while loops with await.

Reply:
Did that. Why not.

Not sure if as a general rule I should be treating XKit 7 PRs as minimum-diff bandaids or clean fixes. Sounds like probably the second?